# Using RandomForest For Classification

## Importing Libraries

In [1]:
import pandas as pd
import numpy as np

## Importing the dataset

In [2]:
dataset = pd.read_csv("data/bill_authentication.csv")
dataset.head()

Variance  Skewness  Curtosis  Entropy  Class
0   3.62160    8.6661   -2.8073 -0.44699      0
1   4.54590    8.1674   -2.4586 -1.46210      0
2   3.86600   -2.6383    1.9242  0.10645      0
3   3.45660    9.5228   -4.0112 -3.59440      0
4   0.32924   -4.4552    4.5718 -0.98880      0

## Dividing data into attributes and labels

In [4]:
X = dataset.iloc[:, 0:4].values
y = dataset.iloc[:, 4].values

## Dividing data into training and testing sets

In [5]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

## Feature Scaling

In [6]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

## Training the Algorithm to solve the Classification problem

In [7]:
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators=20, random_state=0)
regressor.fit(X_train, y_train)
y_pred = regressor.predict(X_test)

## Evaluating the Algorithm by using the metrics

In [16]:
from sklearn.metrics import  classification_report, confusion_matrix, accuracy_score
y_pred = y_pred > 0.7
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))
print(accuracy_score(y_test, y_pred))

[[155   2]
 [  3 115]]
              precision    recall  f1-score   support

           0       0.98      0.99      0.98       157
           1       0.98      0.97      0.98       118

    accuracy                           0.98       275
   macro avg       0.98      0.98      0.98       275
weighted avg       0.98      0.98      0.98       275

0.9818181818181818


### The accuracy achieved for by our random forest classifier with 20 trees is 98.18%